chain/consensus/mir/manager.go

```diff
@@ -234,68 +235,151 @@ func NewManager(ctx context.Context, addr address.Address, h host.Host, api v1ap
 		if err != nil {
 			return nil, fmt.Errorf("failed to create interceptor: %w", err)
 		}
-		m.MirNode, err = mir.NewNode(t.NodeID(mirID), nodeCfg, smrSystem.Modules(), nil, m.interceptor)
+		m.mirNode, err = mir.NewNode(t.NodeID(mirID), nodeCfg, smrSystem.Modules(), nil, m.interceptor)
 	} else {
-		m.MirNode, err = mir.NewNode(t.NodeID(mirID), nodeCfg, smrSystem.Modules(), nil, nil)
+		m.mirNode, err = mir.NewNode(t.NodeID(mirID), nodeCfg, smrSystem.Modules(), nil, nil)
 	}
 	if err != nil {
-		return nil, fmt.Errorf("failed to create Mir node: %w", err)
+		return nil, fmt.Errorf("validator %v failed to create Mir node: %w", mirID, err)
 	}
 
 	return &m, nil
 }
 
-// Start starts the manager.
-func (m *Manager) Start(ctx context.Context) chan error {
-	log.Infof("Mir manager %s starting", m.MirID)
-	log.Info("Mir initial checkpointing period: ", m.StateManager.GetCheckpointPeriod())
+func (m *Manager) Serve(ctx context.Context) error {
+	log.With("validator", m.mirID).Info("Mir manager serve starting")
+	defer log.With("validator", m.mirID).Info("Mir manager serve stopped")
 
-	errChan := make(chan error, 1)
+	log.With("validator", m.mirID).
+		Infof("Mir info:\n\tNetwork - %v\n\tValidator ID - %v\n\tMir peerID - %v\n\tValidators - %v",
+			m.netName, m.mirID, m.mirID, m.initialValidatorSet.GetValidators())
 
+	errChan := make(chan error, 1)
 	go func() {
 		// Run Mir node until it stops.
-		errChan <- m.MirNode.Run(ctx)
+		errChan <- m.mirNode.Run(ctx)
+	}()
+	// Perform cleanup of Node's modules and ensure that mir is closed when we stop mining.
+	defer func() {
+		m.Stop()
 	}()
 
-	return errChan
+	reconfigure := time.NewTicker(ReconfigurationInterval)
```

Comment:
If different validators choose a different `ReconfigurationInterval` would this affect the reconfiguration logic in any way? This is just for local poling purposes , right?

Reply:
That is correct. That should not affect the logic because the validators vote and agree on the next configuration. We do not use synchrony assumptions.